### ISSUE TITLE
Reports using of public property instead of getter/setter

### ISSUE THREAD

**Issue by rentalhost** [kalessil/phpinspectionsea#201]
It should reports if you have a getter/setter over a public property in a simple way.

**For instance:**

```php
class MyClass {
    public $property;

    // Warning: this method could be dropped, read the public $property instead.
    public function getProperty() {
        return $this->property;
    }

    // Warning: this method could be dropped, write on the public $property instead.
    public function setProperty($value) {
        $this->property = $value;
    }
}
```

Note: the method name should not be checked, it could be anything. What matter here is the argument structure and type and the method structure.

**On getter:**

* Method should be public;
* Readed property should be public;
* Returns the own public property crudely;
* Method should not be typehinted (_return type_);

**On setter:**

* Method should be public;
* Wrote property should be public;
* One argument required;
* First argument should not be typehinted;
* Single statement: set the own public property with the first argument crudely;
* Returns void (_no return statement_);
* Method should not be typehinted (_return type_);

**Autofixers:**

* Drop method;
* Change visibility of property to private;

**False-positives:**

The following methods should not be warned.

```php
function getProperty() { return $this->property + 1; }
function getProperty($any) { return $this->property + $any; }
function setProperty() { $this->property = 123; }
function setProperty($any) { return $this->property = $any; }
function setProperty($any = null) { $this->property = $any ?? $this->property; }
function setProperty($a, $b) { $this->property = $a + $b; }
function setProperty(Something $a) { $this->property = $a; }
```

**Comment by funivan** [comment]
One big disadvantage of the public property: it is hard to deprecate them.
You can not just drop public property (for example if you develop library) you need to deprecate it first. 

p.s. I know that there are `__get`, `__set`, `__isset`, `__unset` methods but it is really hard way of public property deprecation

**Comment by funivan** [comment]
@iltar can you explain "should be a blocker for this fixer." ?

**Comment by iltar** [comment]
As you voted with a confused emoji

**Comment by funivan** [comment]
@iltar I get it) If this feature will be interesting for the community - then ok =)

**Comment by funivan** [comment]
Maybe it will be better to just stop writing public properties? =)  
In our team, we avoid using public properties. "Properties should not be public" - this inspection is simple to implement.

**Comment by kalessil** [comment]
Since communities not yet agreed on a common approach, I'll close this ticket an used this time for more architecture inspections.
